### PR TITLE
DLP/WAF integration API changes

### PR DESCRIPTION
**Pull request by ben-taussig-solo**
Co-authored-by: Jacob <57016439+jbohanon@users.noreply.github.com>

# Description

Instantiates a DlpTransformer directly within the ModSecurity filter to ensure sensitive data is masked before audit logs are output.

This PR includes an API change to include DlpTransformation config in the ModSecurity message and a modification of the ModSecurity filter to instantiate a DlpTransformer to run masking rules before audit log generation regardless of filter ordering or direct-response.

BOT NOTES: 
resolves https://github.com/solo-io/solo-projects/issues/3519

### CODE REVIEW

File: changelog/v1.12.0-beta13/dlp-in-waf.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+changelog:
+- type: FIX
+  issueLink: https://github.com/solo-io/solo-projects/issues/3519
+  description: >
+    Instantiates a DlpTransformer directly within the ModSecurity filter to ensure sensitive data is masked before audit logs are output.
```
Comment (kdorosh) [inline changelog/v1.12.0-beta13/dlp-in-waf.yaml:5]: not a great user-facing changelog; i would call this non-user-facing unless we add logic to use and configure this new api to fix user-facing behavior

Comment (kdorosh) [reply changelog/v1.12.0-beta13/dlp-in-waf.yaml:5]: lets call this non user facing since the impl is in enterprise, so the fix is in enterprise

Comment (ben-taussig-solo) [reply changelog/v1.12.0-beta13/dlp-in-waf.yaml:5]: Makes sense, resolved here: https://github.com/solo-io/gloo/pull/6443/commits/765798cd54ebadb8e64557a2b0046d4cba911ffa

**Review by elcasteel — DISMISSED** [review]

File: projects/gloo/api/external/envoy/extensions/waf/waf.proto [hunk 2]
```diff
@@ -51,6 +53,8 @@ message ModSecurity {
   // this format is a multiline log format, so it is disabled for regular use.
   // do not enable this in production!
   bool regression_logs = 4;
+  // DLP Transformation to perform at ModSecurity processing time
```
Comment (nfuden) [inline projects/gloo/api/external/envoy/extensions/waf/waf.proto:56]: nit: This means almost nothing over the basic line. While its good to not leak implementation details I would like to either not have the comment or have it not just be jargon.

File: projects/gloo/api/external/envoy/extensions/waf/waf.proto [hunk 3]
```diff
@@ -77,4 +81,6 @@ message ModSecurityPerRoute {
   // This can help improve performance.
   bool request_headers_only = 6;
   bool response_headers_only = 7;
+  // DLP Transformation to perform at ModSecurity processing time
```
Comment (nfuden) [inline projects/gloo/api/external/envoy/extensions/waf/waf.proto:84]: nit: This means almost nothing over the basic line. While its good to not leak implementation details I would like to either not have the comment or have it not just be jargon.

File: projects/gloo/api/external/envoy/extensions/waf/waf.proto (hunk 2 above)
Comment (ben-taussig-solo) [reply projects/gloo/api/external/envoy/extensions/waf/waf.proto]: Removed the comment here: 88e2a48c9a9b75bc98f3b436f863ff9d51572ac0

File: projects/gloo/api/external/envoy/extensions/waf/waf.proto (hunk 3 above)
Comment (ben-taussig-solo) [reply projects/gloo/api/external/envoy/extensions/waf/waf.proto]: Removed the comment here: 88e2a48c9a9b75bc98f3b436f863ff9d51572ac0

**Review by nfuden — APPROVED** [review]

**Review by nfuden — APPROVED** [review]

**Pull request merged**